### PR TITLE
IDEX-2886: added method to read specific file content to machine API

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 1]
```diff
@@ -803,6 +790,36 @@ private LineConsumer getLogger(LineConsumer fileLogger, String outputChannel) th
         return fileLogger;
     }
 
+    private void exec(final InstanceProcess instanceProcess, final String machineId, final LineConsumer processLogger) {
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: document this

Comment (skabashnyuk) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: +

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java [hunk 2]
```diff
@@ -331,6 +334,24 @@ public void getProcessLogs(@PathParam("machineId") String machineId,
...
+    @Path("/{machineId}/filepath/{path:.*}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @RolesAllowed("user")
+    public String getFileContent(@PathParam("machineId") String machineId,
+                                 @PathParam("path") String path)
+            throws NotFoundException, ForbiddenException, ServerException, IOException {
+        checkCurrentUserPermissions(machineManager.getMachine(machineId));
+
+        CommandDescriptor descriptor = newDto(CommandDescriptor.class)
+                .withCommandLine(String.format("cat %1$2s", path));
```
Comment (evoevodin) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java]: i think you can define it in one line and without of external formatting 
<code> CommandDescriptor descriptor = newDto(CommandDescriptor.class).withCommandLine("cat " + path)</code>

Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java]: +1

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 3]
```diff
@@ -582,42 +582,28 @@ public void removeSnapshots(String owner, String workspaceId, ProjectBinding pro
      */
     public InstanceProcess exec(final String machineId, final Command command, String clientOutputChannel)
             throws NotFoundException, MachineException {
+        return exec(machineId, command, clientOutputChannel, null, true);
+    }
+
+    public InstanceProcess exec(final String machineId,
+                                final Command command,
+                                final String clientOutputChannel,
```
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: why this method contains both processLogger and clientOutputChannel?
I propose separate it into 2 methods. First accepts only process logger. Another one accepts only client output channel, creates process logger and calls the first one.
Wdyt?

**Pull request closed without merging**